controllers/argocd/configmap.go

```diff
@@ -390,10 +390,17 @@ func (r *ReconcileArgoCD) reconcileArgoConfigMap(cr *argoprojv1a1.ArgoCD) error
 
 	existingCM := &corev1.ConfigMap{}
 	if argoutil.IsObjectFound(r.Client, cr.Namespace, cm.Name, existingCM) {
+		// This logic should be refactored as part of SSO Unification.
+		// https://github.com/argoproj-labs/argocd-operator/pull/646
+		// In future, Both Keycloak and Dex can be configured using `.spec.SSO`.
 		if cr.Spec.SSO == nil {
 			if err := r.reconcileDexConfiguration(existingCM, cr); err != nil {
 				return err
 			}
+		} else {
+			if cr.Spec.SSO.Provider == "keycloak" {
```

Comment:
I feel the need for constants here for terms like "keycloak" and "dex"

Reply:
Sure @jannfis , totally makes sense. I will add the required.

Reply:
There is already one such. Pushed a new commit @jannfis